setup_and_run.sh

```diff
@@ -0,0 +1,238 @@
+    run_api "$workers" "$port"
+    run_scheduler
+
+    log "Setup and deployment completed successfully!"
+    log "API running on port $port with $workers workers"
+    log "Check ./api.log and ./scheduler.log for details"
+}
+restart() {
+    local workers=${1:-8}
+    local port=${2:-8081}
+    local path=${3:-"/nearai"}
+
+    log "Restarting processes without installation..."
+
+    # Check directory
+    check_directory "$path"
+
+    # Activate virtual environment if exists
+    if [[ -d "venv" ]]; then
+        log "Activating virtual environment..."
+        source venv/bin/activate
+    else
+        log "Error: Virtual environment 'venv' not found. Please run setup first."
+        exit 1
+    fi
+
+    cd hub
+
+    # Stop all processes
+    stop_all
+
+    # Start API and scheduler
+    run_api "$workers" "$port"
+    run_scheduler
+
+    log "Restart completed successfully!"
+    log "API running on port $port with $workers workers"
+    log "Check ./api.log and ./scheduler.log for details"
+}
+# Help message
+show_help() {
+    echo "Usage: ./setup_and_run.sh [command] [options]"
+    echo ""
+    echo "Commands:"
+    echo "  [workers] [port] [path]  Start the application with specified parameters (default)"
+    echo "  stop                     Stop all running processes"
+    echo "  restart [workers] [port] [path]  Restart processes without reinstallation"
+    echo "  --help, -h               Show this help message"
+    echo ""
+    echo "Options:"
+    echo "  workers     Number of worker processes (default: 8)"
+    echo "  port        Port to run the API on (default: 8081)"
+    echo "  path        Directory path (default: /nearai)"
+    echo ""
+    echo "Examples:"
+    echo "  ./setup_and_run.sh                        # Start with default settings"
+    echo "  ./setup_and_run.sh 4 8000                 # Start with 4 workers on port 8000"
+    echo "  ./setup_and_run.sh restart 4 8000         # Restart with custom workers and port"
+    echo "  ./setup_and_run.sh stop                   # Stop all processes"
+
+}
+# Stop all processes
+stop_all() {
+    log "Stopping API and scheduler processes..."
+
+    # Stopping API process
+    if [ -f api.pid ]; then
+        local api_pid=$(cat api.pid)
+        if kill -0 "$api_pid" 2>/dev/null; then
+            kill "$api_pid" 2>/dev/null
+            log "Sent termination signal to API process (PID: $api_pid)"
+            sleep 1  # Wait for process to terminate
+            if kill -0 "$api_pid" 2>/dev/null; then
+                kill -9 "$api_pid" 2>/dev/null
+                log "Forced termination of API process (PID: $api_pid)"
+            fi
+        else
+            log "API process (PID: $api_pid) not running"
+        fi
+        rm api.pid
+    fi
+
+    # If the API process is still running (PID didn't work or file wasn't there), kill by name
+    pkill -f "nearai-hub-worker" || true
+    pkill -f "multiprocessing.spawn" || true
```

Comment:
At this point it feel like we're reimplementing `systemd` from scratch.  WDYT about moving over to `systemd`?  We could keep this script but it would get a lot simpler, just pip install and `systemctl service restart ...`.

Reply:
`hub service` pr was existed some time ago https://github.com/nearai/nearai/pull/117/files we can reopen it

this script was created mostly to setup a hub after the update